go/cmd/dolt/commands/admin/archive.go

```diff
@@ -116,8 +124,77 @@ func (cmd ArchiveCmd) Exec(ctx context.Context, commandStr string, args []string
 	return 0
 }
 
+func handleProgress(ctx context.Context, progress chan interface{}) {
+	go func() {
+		rotation := 0
+		p := cli.NewEphemeralPrinter()
+		currentMessage := "Starting Archive Build"
+		var lastProgressMsg *nbs.ArchiveBuildProgressMsg
+		lastUpdateTime := time.Now()
+
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case msg, ok := <-progress:
+				if !ok {
+					return
+				}
+				switch v := msg.(type) {
+				case string:
+					cli.Printf("LOG: %s\n", v)
+				case nbs.ArchiveBuildProgressMsg:
+					if v.Total == v.Completed {
+						p.Printf("%s: Done\n", v.Stage)
+						lastProgressMsg = nil
+						currentMessage = ""
+						p.Display()
+						cli.Printf("\n")
+					} else {
+						lastProgressMsg = &v
+					}
+				default:
+					cli.Printf("Unexpected Message: %v\n", v)
+				}
+			case <-time.After(1 * time.Second):
```

Comment:
Is this line doing anything? I couldn't figure it out 😄

Reply:
Ahh... it's to trigger displaying a progress update, even if no progress event comes in? Might be worth a short comment in the case block mentioning the fallthrough to that logic, or maybe even pulling the `if` block below out to a helper function and calling it there if you wanted to be fancy. Not a big deal either way, just wasn't super obvious on my first read.